src/mimir_cluster.py

```diff
@@ -1,39 +1,106 @@
-"""This file defines the schemas for the provider and requirer sides of this relation interface.
+#!/usr/bin/env python3
+# Copyright 2024 Canonical
+# See LICENSE file for licensing details.
 
-It must expose two interfaces.schema_base.DataBagSchema subclasses called:
-- ProviderSchema
-- RequirerSchema
+"""This module contains an endpoint wrapper class for the requirer side of the ``mimir-cluster`` relation.
 
-TODO: see https://github.com/canonical/charm-relation-interfaces/issues/121
+As this relation is cluster-internal and not intended for third-party charms to interact with `mimir-coordinator-k8s`, its only user will be the mimir-worker-k8s charm. As such, it does not live in a charm lib as most other relation endpoint wrappers do.
 """
+
 import json
 import logging
-from collections import defaultdict
 from enum import Enum
-from typing import Any, Dict, MutableMapping, Set, List, Iterable
-from typing import Optional
+from typing import Any, Dict, Iterable, List, MutableMapping, Optional, Set
 from urllib.parse import urlparse
 
 import ops
 import pydantic
-from ops import Object, ObjectEvents, EventSource, RelationCreatedEvent
+from ops import EventSource, Object, ObjectEvents, RelationCreatedEvent
 from pydantic import BaseModel, ConfigDict
 
 log = logging.getLogger("mimir_cluster")
-LIBID = "9818a8d44028454a94c6c3a01f4316d2"
-DEFAULT_ENDPOINT_NAME = "mimir-cluster"
-
-LIBAPI = 0
-LIBPATCH = 3
 
+DEFAULT_ENDPOINT_NAME = "mimir-cluster"
 BUILTIN_JUJU_KEYS = {"ingress-address", "private-address", "egress-subnets"}
-
 MIMIR_CONFIG_FILE = "/etc/mimir/mimir-config.yaml"
 MIMIR_CERT_FILE = "/etc/mimir/server.cert"
 MIMIR_KEY_FILE = "/etc/mimir/private.key"
 MIMIR_CLIENT_CA_FILE = "/etc/mimir/ca.cert"
 
 
+class MimirRole(str, Enum):
```

Comment:
Should we drop `Mimir` from the class name?

I haven't used enums in python before. From the [docs](https://docs.python.org/3/howto/enum.html) it seems like:
- We should `from enum import unique` and use that decorator here.
- The `META_ROLES, expand_roles` below can be static/classmethods on the enum itself.
- Add TODO to switch to [`enum.StrEnum`](https://docs.python.org/3/library/enum.html#enum.StrEnum) when jammy is no longer supported :sweat_smile: